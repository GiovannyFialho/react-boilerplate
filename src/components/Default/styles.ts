import styled from "styled-components";
import media from "styled-media-query";

export const Wrapper = styled.div`
    background-color: #fafafa;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100vh;
    padding: 0 2rem;
`;

export const Title = styled.h2`
    font-size: 7rem;
    text-align: center;
    margin-bottom: 5rem;
    background: linear-gradient(to right, #51af57 1%, #68e170 100%);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    ${media.lessThan("small")`
        font-size: 5rem;
    `}
    ${media.greaterThan("medium")`
        font-size: 15rem;
        margin-bottom: 10rem;
    `}
`;

export const Techs = styled.div`
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
`;

export const Icon = styled.div`
    width: 6rem;
    margin: 3rem;
    &::last-child {
        margin-right: 0;
    }
    ${media.lessThan("small")`
        width: 4rem;
        margin: 1rem;
    `}
    ${media.greaterThan("medium")`
        width: 7rem;
    `}
    svg {
        color: #68e170;
    }
`;
