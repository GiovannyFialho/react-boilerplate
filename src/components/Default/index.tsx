import {
    Typescript,
    StyledComponents,
    Jest,
    Storybook,
    NextDotJs
} from "@styled-icons/simple-icons";
import { ReactLogo } from "@styled-icons/boxicons-logos/ReactLogo";

import { Wrapper, Title, Techs, Icon } from "./styles";

type Tech =
    | "typescript"
    | "reactJS"
    | "nextJS"
    | "styledComponents"
    | "jest"
    | "storybook";

export type DefaultProps = {
    title: string;
    techs?: Tech[];
};

const Default = ({ title, techs }: DefaultProps) => {
    const techIcons = {
        typescript: <Typescript title="typescript" />,
        reactJS: <ReactLogo title="React JS" />,
        nextJS: <NextDotJs title="Next JS" />,
        styledComponents: <StyledComponents title="Styled Components" />,
        jest: <Jest title="Jest" />,
        storybook: <Storybook title="Storybook" />
    };

    return (
        <Wrapper>
            <Title>{title}</Title>

            {techs && (
                <Techs>
                    {techs.map((tech, index) => (
                        <Icon key={index} role="image" title={tech}>
                            {techIcons[tech]}
                        </Icon>
                    ))}
                </Techs>
            )}
        </Wrapper>
    );
};

export default Default;
