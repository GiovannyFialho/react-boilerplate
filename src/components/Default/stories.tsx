import { Meta, Story } from "@storybook/react/types-6-0";

import Default, { DefaultProps } from ".";

export default {
    title: "Default",
    component: Default,
    parameters: {
        layout: "fullscreen"
    },
    argTypes: {
        techs: {
            control: {
                type: "inline-check",
                options: [
                    "typescript",
                    "reactJS",
                    "nextJS",
                    "styledComponents",
                    "jest",
                    "storybook"
                ]
            }
        }
    }
} as Meta;

export const Basic: Story<DefaultProps> = (args) => <Default {...args} />;

Basic.args = {
    title: "React Boilerplate",
    techs: [
        "typescript",
        "reactJS",
        "nextJS",
        "styledComponents",
        "jest",
        "storybook"
    ]
};
