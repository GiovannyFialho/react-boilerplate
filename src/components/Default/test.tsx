import { render, screen } from "@testing-library/react";

import Default, { DefaultProps } from ".";

const props: DefaultProps = {
    title: "React Boilerplate",
    techs: ["typescript"]
};

describe("<Default />", () => {
    it("should render component", () => {
        const { container } = render(<Default {...props} />);

        expect(
            screen.getByRole("heading", { name: /React Boilerplate/i })
        ).toBeInTheDocument();

        expect(
            screen.getByRole("image", { name: /typescript/i })
        ).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });
});
