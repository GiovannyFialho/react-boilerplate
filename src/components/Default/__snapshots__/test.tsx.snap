// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Default /> should render component 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  background-color: #fafafa;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  padding: 0 2rem;
}

.c1 {
  font-size: 7rem;
  text-align: center;
  margin-bottom: 5rem;
  background: linear-gradient(to right,#51af57 1%,#68e170 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 {
  width: 6rem;
  margin: 3rem;
}

.c3::last-child {
  margin-right: 0;
}

.c3 svg {
  color: #68e170;
}

@media (max-width:450px) {
  .c1 {
    font-size: 5rem;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 15rem;
    margin-bottom: 10rem;
  }
}

@media (max-width:450px) {
  .c3 {
    width: 4rem;
    margin: 1rem;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 7rem;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    React Boilerplate
  </h2>
  <div
    class="c2"
  >
    <div
      class="c3"
      role="image"
      title="typescript"
    >
      <svg
        class="c4"
        fill="currentColor"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          typescript
        </title>
        <path
          d="M1.125 0C.502 0 0 .502 0 1.125v21.75C0 23.498.502 24 1.125 24h21.75c.623 0 1.125-.502 1.125-1.125V1.125C24 .502 23.498 0 22.875 0zm17.363 9.75c.612 0 1.154.037 1.627.111a6.38 6.38 0 0 1 1.306.34v2.458a3.95 3.95 0 0 0-.643-.361 5.093 5.093 0 0 0-.717-.26 5.453 5.453 0 0 0-1.426-.2c-.3 0-.573.028-.819.086a2.1 2.1 0 0 0-.623.242c-.17.104-.3.229-.393.374a.888.888 0 0 0-.14.49c0 .196.053.373.156.529.104.156.252.304.443.444s.423.276.696.41c.273.135.582.274.926.416.47.197.892.407 1.266.628.374.222.695.473.963.753.268.279.472.598.614.957.142.359.214.776.214 1.253 0 .657-.125 1.21-.373 1.656a3.033 3.033 0 0 1-1.012 1.085 4.38 4.38 0 0 1-1.487.596c-.566.12-1.163.18-1.79.18a9.916 9.916 0 0 1-1.84-.164 5.544 5.544 0 0 1-1.512-.493v-2.63a5.033 5.033 0 0 0 3.237 1.2c.333 0 .624-.03.872-.09.249-.06.456-.144.623-.25.166-.108.29-.234.373-.38a1.023 1.023 0 0 0-.074-1.089 2.12 2.12 0 0 0-.537-.5 5.597 5.597 0 0 0-.807-.444 27.72 27.72 0 0 0-1.007-.436c-.918-.383-1.602-.852-2.053-1.405-.45-.553-.676-1.222-.676-2.005 0-.614.123-1.141.369-1.582.246-.441.58-.804 1.004-1.089a4.494 4.494 0 0 1 1.47-.629 7.536 7.536 0 0 1 1.77-.201zm-15.113.188h9.563v2.166H9.506v9.646H6.789v-9.646H3.375z"
        />
      </svg>
    </div>
  </div>
</div>
`;
